plugins/GrapesJsBuilderBundle/Assets/library/js/builder.service.js

```diff
@@ -97,14 +97,34 @@ export default class BuilderService {
         data: { filename: response.getFilename() },
       });
     });
-
   }
 
   /**
    * Initialize the grapesjs build in the
    * correct mode
    */
   initGrapesJS(object) {
+    // grapesjs-custom-plugins: add globally defined mautic-grapesjs-plugins using name as pluginId for the plugin-function
+    if (window.MauticGrapesJsPlugins) {
+      window.MauticGrapesJsPlugins.forEach((item) => {
+        let error = false;
+
+        if (!item.name) {
+          console.warn('A name is required for Mautic-GrapesJs plugins in window.MauticGrapesJsPlugins. Registration skipped!');
+          error = true;
```

Comment:
If this (and row 119) would be replaced with `return;` then the complexity of this function could be reduced by 1 if statement. Please consider if it's worth changing.

Reply:
You mean, that the loop should check first only for the plugin-name (and return with warning if it hasn't one), and in a second run checking for a function (and return with warning it it isn't one)? Than yes, we could do that, but I thought checking both at the same time makes sense. Up to you guys, please let me know if I should change that.

Reply:
It's fine to check both in the same time. The `forEach` function is getting a callback function as an argument. In that callback you want to either throw warnings for debugging or call `grapesjs.plugins.add(item.name, item.plugin)`. So you may as well return early after each warning so it won't get to adding the plugin.

Reply:
When using code instead of words:
```js
window.MauticGrapesJsPlugins.forEach((item) => {
        let error = false;

        if (!item.name) {
          console.warn('A name is required for Mautic-GrapesJs plugins in window.MauticGrapesJsPlugins. Registration skipped!');
          return;
        }

        if (typeof item.plugin !== 'function') {
          console.warn('The Mautic-GrapesJs plugin must be a function in window.MauticGrapesJsPlugins. Registration skipped!');
          return;
        }
        grapesjs.plugins.add(item.name, item.plugin);
      });
    }
```

Reply:
Yes, now in "worst-case" a dev needs two runs, first fixing the name and in a second one giving a proper function... but ok, its not really a worse case^^ Have changed it.